src/components/learner-credit-management/cards/data/useCourseCardMetadata.jsx

```diff
@@ -54,10 +55,20 @@ const useCourseCardMetadata = ({
     title,
     courseRuns,
   } = course;
+  const {
+    dataByContentKey: catalogContainsRestrictedRunsData,
+    isLoading: isLoadingcatalogContainsRestrictedRuns,
+  } = useCatalogContainsContentItemsMultipleQueries(
+    subsidyAccessPolicy.catalogUuid,
+    // Pass only restricted runs.
+    courseRuns?.filter(run => run.restrictionType === 'custom-e2e-enterprise'), // TODO: replace with constant
```

Comment:
Should this be passing the course run **keys** vs. the entire course run objects? It appears the entire course run object is added in the query key vs. just the course run `key`.

Reply:
nit: I also wonder if it's worth abstracting a utility function for this filtering for restricted runs, given the same filtering logic is used in 2 places?

Reply:
leaving open to remind myself to implement more DRYness on filtering restricted runs.